### PR TITLE
Reduce schema reuse and promote dependencies to top level directory [Part 2]

### PR DESCRIPTION
**Pull request by jvc9109**
## Summary
The goal is to reduce excessive schema reuse in the scope of cleaning up the api-definitions files.
Cleaned subschemas (not referenced from the paths ) are moved to top level directory.

## Links
[Part 1](https://github.com/Rebilly/api-definitions/pull/1476)

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/requestBodies/CreditTransactionRequest.yaml [hunk 1]
```diff
@@ -1,6 +1,124 @@
...
+          type: string
+          maxLength: 255
+        notificationUrl:
+          nullable: true
+          description: |-
+            URL where a server-to-server `POST` notification is sent.
+            This notification is sent when the transaction result is finalized after a timeout or an offsite interaction.
+
+            Do not trust this notification alone,
+            complete a `GET` request to confirm the result of the transaction.
```
Comment (noon-dawg) [inline openapi/components/requestBodies/CreditTransactionRequest.yaml:82]: nitpick: 'trust this notification alone' is ambiguous 
```suggestion
            Do not interpret this notification as a confirmation,
            complete a `GET` request to confirm the result of the transaction.
```

File: openapi/components/requestBodies/TransactionRequest.yaml [hunk 2]
```diff
@@ -1,6 +1,147 @@
...
+          type: string
+          maxLength: 255
+        notificationUrl:
+          nullable: true
+          description: |-
+            URL where a server-to-server `POST` notification is sent.
+            This notification is sent when the transaction result is finalized after a timeout or an offsite interaction.
+
+            Do not trust this notification alone,
+            complete a `GET` request to confirm the result of the transaction.
```
Comment (noon-dawg) [inline openapi/components/requestBodies/TransactionRequest.yaml:111]: nitpick: 'trust this notification alone' is ambiguous 
```suggestion
            Do not interpret this notification as a confirmation,
            complete a `GET` request to confirm the result of the transaction.
```

**Review by noon-dawg — COMMENTED** [review]
Looks good on the docs side

File: openapi/components/requestBodies/CreditTransactionRequest.yaml [hunk 3]
```diff
@@ -1,6 +1,124 @@
...
+            URL where a server-to-server `POST` notification is sent.
+            This notification is sent when the transaction result is finalized after a timeout or an offsite interaction.
+
+            Do not interpret this notification as a confirmation,
+            complete a `GET` request to confirm the result of the transaction.
+            To ensure the request is not reattempted,
+            when the result is confirmed, respond with a `2xx` HTTP status code.
+
+            The following placeholders are available to use in this URI: `{id}` and `{result}`.
+            These placeholders are replaced the with the transaction ID and result accordingly.
```
Comment (2e3s) [inline openapi/components/requestBodies/CreditTransactionRequest.yaml:87]: Copies of a structures is understandable and maintenable, but wouldn't copies of free-form description texts be much more prone to desynchronisation without some ? Although, I'm not sure about the proper solution.

**Review by 2e3s — APPROVED** [review]

File: openapi/components/requestBodies/CreditTransactionRequest.yaml (hunk 3 above)
Comment (jvc9109) [reply openapi/components/requestBodies/CreditTransactionRequest.yaml:87]: For this particular case, `schemas/Transactions/CreditTransactionRequest.yaml` was only used in this request body. So there is no actual re-use for that schema

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**